enhancements/subscription-content/subscription-injection.md

```diff
@@ -0,0 +1,397 @@
+2. Extract the `redhat.repo` file from the UBI8 image and add it to the `openshift-config` namespace
+   as the `redhat-repo-rhel8` `ConfigMap`.
+3. Download an OpenShift entitlement key set from the customer portal, and add it to the
+   `openshift-config` namespace as the `etc-pki-entitlement` `Secret`.
+4. Run a pod with the `"subscription.openshift.io/cluster-bundle": rhel8` annotation, and a UBI8
+   container which performs a `yum install` of a RHEL package that requires a subscription.
+
+### Graduation Criteria
+
+This feature will likely require Dev Preview, Tech Preview, and GA maturity levels.
+
+#### Dev Preview
+
+- Cluster-level APIs at `v1alpha1`
+- Operator configuration APIs at `v1alpha1`
+- Dev preview release of the projected resource CSI driver via OLM (as community operator)
+- Release as a community OLM operator
+
+##### Dev Preview -> Tech Preview
+
+- Projected resource CSI driver reaches tech preview state
+- Cluster-level API reaches `v1beta1`
+- Clear documentation on how to enable subscription injection across the cluster
+- Initial scale testing
+- Gather feedback for namespace-restricted subscription bundles
+- Release as a tech preview, official OLM operator
+
+##### Tech Preview -> GA
+
+- All APIs reach `v1` stability
+- Projected resource CSI driver reaches GA
+- Security audit for these components and the projected resource CSI driver
+- Scale testing (with potential mitigations like `HorizontalPodAutoscalers`)
+- Disruption testing (with potential mitigations like `PodDisruptionBudgets`)
+
+### Upgrade / Downgrade Strategy
+
+Upgrades and downgrades will be handled via OLM. Each released version will specify:
+
+- Minimum supported version of the projected resource CSI driver
+- Minimum supported kube version of the cluster.
+
+### Version Skew Strategy
+
+How will the component handle version skew with other components? What are the guarantees? Make sure
+this is in the test plan.
+
+Consider the following in developing a version skew strategy for this enhancement:
+
+- During an upgrade, we will always have skew among components, how will this impact your work?
+- Does this enhancement involve coordinating behavior in the control plane and in the kubelet? How
+  does an n-2 kubelet without this feature available behave when this feature is used?
+- Will any other components on the node change? For example, changes to CSI, CRI or CNI may require
+  updating that component before the kubelet.
+
+## Implementation History
+
+Major milestones in the life cycle of a proposal should be tracked in `Implementation History`.
+
+## Drawbacks
+
+- Since this is being deliverd via OLM, customers will need to opt into capabilities that they may
+  expect by default.
+- Users will have to specify on a per-workload basis which subscriptions should be added, increasing
+  toil.
+- Cluster admins need to manually obtain the entitlement keys and `redhat.repo` configurations.
+
+## Alternatives
+
```

Comment:
```suggestion

### Create a privileged DaemonSet that manages the host subscriptions

Such a DaemonSet wouldn't need to be fully privileged; only it would need
access to `/etc/pki/{entitlement,consumer}` and `/etc/rhsm` mounts on the host.

This would have the benefit of "it just works" without any further
configuration from the end user to use the entitlements on RHEL 6, 7, or 8
containers. Likewise for UBI 7 and 8 containers.
```

Reply:
I'll add this as an alternative in #384.

Reply:
With this alternative, how would we handle RHCOS nodes which do not have rhsm installed? What would the dameonset need to do when the node is drained and removed from the cluster? Likewise when the node joins the cluster via an auto-scaling mechanism.